trading/trader/orders_manager.py

```diff
@@ -111,53 +114,51 @@ def set_order_refresh_time(self, seconds):
     # TODO : currently blocking, may implement queue if needed
-    def force_update_order_status(self, blocking=True, simulated_time=False):
+    async def force_update_order_status(self, blocking=True, simulated_time=False):
         if blocking:
-            self._update_orders_status(simulated_time=simulated_time)
+            await self._update_orders_status(simulated_time=simulated_time)
         else:
             raise NotImplementedError("force_update_order_status(blocking=False) not implemented")
 
-    def _update_orders_status(self, simulated_time=False):
+    async def _update_orders_status(self, simulated_time=False):
         """
         Prepare order status updating by getting price data
         then ask orders to check their status
         Finally ask cancellation and filling process if it is required
         """
 
         # update all prices
-        self._update_last_symbol_list(True)
+        await self._update_last_symbol_list(True)
 
         for order in copy.copy(self.order_list):
-            # symbol prices from exchange
-            if order.get_order_symbol() in self.last_symbol_prices:
-                with order as odr:
-                    odr.set_last_prices(self.last_symbol_prices[odr.get_order_symbol()])
 
             # ask orders to update their status
-            with order as odr:
+            async with order.get_lock():
+                odr = order
+                # update symbol prices from exchange
+                if odr.get_order_symbol() in self.last_symbol_prices:
+                    odr.set_last_prices(self.last_symbol_prices[odr.get_order_symbol()])
+
                 if odr in self.order_list:
-                    odr.update_order_status(simulated_time=simulated_time)
+                    await odr.update_order_status(simulated_time=simulated_time)
 
                     if odr.get_status() == OrderStatus.FILLED:
                         self.logger.info(f"{odr.get_order_symbol()} {odr.get_name()} (ID : {odr.get_id()})"
                                          f" filled on {self.trader.get_exchange().get_name()} "
                                          f"at {odr.get_filled_price()}")
-                        odr.close_order()
+                        await odr.close_order()
 
-    async def start(self):
+    async def poll_update(self):
         """
         Async method that will periodically update orders status with update_orders_status
+        Should never be called in backtesting
         """
-
-        if Backtesting.enabled(self.config):
-            self.keep_running = False
         while self.keep_running:
             try:
                 # call update status
-                self._update_orders_status()
+                await self._update_orders_status()
             except Exception as e:
                 self.logger.error("Error when updating orders")
                 self.logger.exception(e)
-                await asyncio.sleep(self.order_refresh_time)
```

Comment:
Why removed ? It prevent from spamming isn't it ?

Reply:
It was supposed to but we still call await asyncio.sleep right after so i though it with have become obsolete, if you think in an exception case it's worth waiting 2x the normal time, we can put it back